### PR TITLE
Use AWS SDK v3, specify imports, allow passing in `max-wait-time`

### LINKED ISSUE #7: Allow user-provided timeout
When containers don't come up healthily in Fargate, ECS doesn't tell CodeDeploy that something's wrong. So, sometimes, we get deployments that wait for 60 minutes while ECS thrashes. With a lower timeout, developers could get faster feedback without needing to check how ECS is doing.

### PR DESCRIPTION
**Pull request by GaryGSC**
This should remove the deprecation warning:
image

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,94 +1,78 @@
...
 
 Link to deployment: ${shortcutLink}`)
 
-    await codeDeploy
-      .waitFor('deploymentSuccessful', {
-        deploymentId: deployment.deploymentId
-      })
-      .promise()
+    const maxWaitTime = 3600 // One hour
+    await waitUntilDeploymentSuccessful({ client: codedeployClient, maxWaitTime }, { deploymentId })
```
Comment (GaryGSC) [inline src/index.ts:51]: This was interesting. The new version had `maxWaitTime` as a required field. I believe v2 defaulted to 1 hour, so I used that.

Comment (GaryGSC) [reply src/index.ts]: Edit: We now allow users to pass in a value, but still default to one hour.

File: yarn.lock
Comment (rhettjay) [inline yarn.lock:1]: This is unrelated so feel free to ignore, but do we need yarn?

**Review by rhettjay — APPROVED** [review]

File: yarn.lock
Comment (GaryGSC) [reply yarn.lock:1]: IMO, no. Looks like it was introduced in https://github.com/byu-oit/github-action-codedeploy/commit/6008638e53854d62b9213ff4fd7b06c40c772ce2, by @yoshutch.

We're not using a monorepo, which is usually the reason I hear about people switching to Yarn (though that's less of a thing now that npm workspaces exist). And if we were concerned about install speed, these days, that'd usually imply switching to Bun.

**Pull request merged**